### PR TITLE
feat(pgwire):support binary format param of timestampz and interval type 

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- support binary format param of timestamptz and interval 
- support binary format result of timestamptz 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#3676

### CODE REVIEW

**Review by BowenXiao1999 — COMMENTED** [review]
Most LGTM. @xiangjinwu PTAL.

1. can add some e2e test or unit test?
2. Timestampz -> Timestamptz, why?

File: src/expr/src/vector_op/cast.rs [hunk 1]
```diff
@@ -115,6 +117,17 @@ pub fn timestampz_to_utc_string(elem: i64) -> String {
     instant.format("%Y-%m-%d %H:%M:%S%.f%:z").to_string()
 }
 
+pub fn timestampz_to_utc_binary(elem: i64) -> Bytes {
+    // Just a meaningful representation as placeholder. The real implementation depends on TimeZone
+    // from session. See #3552.
+    let instant = Utc.timestamp_nanos(elem * 1000);
+    let mut out = BytesMut::new();
+    instant
+        .to_sql(&postgres_types::Type::ANY, &mut out)
```
Comment (xiangjinwu) [inline src/expr/src/vector_op/cast.rs:126]: Why not `postgres_types::Type::TIMESTAMPTZ`?

**Review by xiangjinwu — APPROVED** [review]
LGTM, and thanks for the timely support!

As Bowen said above, can we have some e2e tests?

**Comment by ZENOTME** [comment]
> Most LGTM. @xiangjinwu PTAL.
> 
> 1. can add some e2e test or unit test?
> 2. Timestampz -> Timestamptz, why?

1. I will investigate and add e2e test later 
2. because postgres call this type [timestamptz](https://www.postgresql.org/docs/current/datatype-datetime.html#:~:text=timestamptz%20is%20accepted%20as%20an%20abbreviation%20for%20timestamp%20with%20time%20zone%3B%20this%20is%20a%20PostgreSQL%20extension.)😂, I'm not sure whether exist a type called timestampz(But seems we use 'timestampz' across the whole project

File: src/expr/src/vector_op/cast.rs (hunk 1 above)
Comment (ZENOTME) [reply src/expr/src/vector_op/cast.rs:126]: Did you say &postgres_types::Type::ANY? This is only a placeholder. I will add a comment later.

Comment (xiangjinwu) [reply src/expr/src/vector_op/cast.rs:126]: Yes> I am curious why we pass `ANY` here

Comment (ZENOTME) [reply src/expr/src/vector_op/cast.rs:126]: This place needs a reference of postgres_types::Type, I just pick one postgres_types::Type casually 😂

File: src/utils/pgwire/src/pg_extended.rs [hunk 2]
```diff
@@ -804,5 +806,25 @@ mod tests {
                 "'2021-01-07 06:13:20'::TIMESTAMP"
             ]
         );
+
+        let mut raw_params = vec![BytesMut::new(); 2];
```
Comment (BowenXiao1999) [inline src/utils/pgwire/src/pg_extended.rs:810]: We can add some comments to indicate what feature we are testing or abstract into some small test function with meaningful name.

**Review by BowenXiao1999 — APPROVED** [review]

**Pull request merged**